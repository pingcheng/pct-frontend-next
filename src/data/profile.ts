export const profile = {
  fullName: "Ping Cheng",
} as const;

export const socialUrls = {
  github: "https://github.com/pingcheng",
} as const;
