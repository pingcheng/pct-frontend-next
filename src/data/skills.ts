export const backendStack: string[] = [
  "Node, TypeScript, JavaScript",
  "Java, ColdFusion",
  "PHP, Laravel, Moodle",
  "MySQL, MSSQL, Redis",
  "Server management, Message Queue",
  "Payment Integration",
];

export const frontendStack: string[] = [
  "HTML",
  "JavaScript, Vue, React, jQuery",
  "CSS, SASS, TailwindCSS, Styled Components, Bootstrap",
  "Webpack",
];

export const devOpsStack: string[] = [
  "AWS",
  "Infrastructure planning and implementing",
  "Horizontal scaling",
  "Serverless, Docker, ECS",
  "CI/CD, GitlabCI, CircleCI",
];
