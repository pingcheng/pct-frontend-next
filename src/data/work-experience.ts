export type WorkPosition = {
  position: string;
  startDate: string;
  endDate: string;
  description: string[];
};

export type Company = {
  name: string;
  linkedInUrl?: string;
};

export type WorkExperience = {
  company: Company;
  positions: WorkPosition[];
};

export const workExperiences: WorkExperience[] = [
  {
    company: {
      name: "REA Group - PropTrack",
      linkedInUrl: "https://www.linkedin.com/company/rea-group",
    },
    positions: [
      {
        position: "Lead Developer",
        startDate: "Aug 2024",
        endDate: "Present",
        description: [],
      },
      {
        position: "Senior Developer",
        startDate: "Jul 2022",
        endDate: "Aug 2024",
        description: [
          "Highly involved in digital mortgage areas, seamlessly connecting banks and the valuation industry with modern digital solutions. Contributing to the development of secure, enterprise-level products.",
          ":line-break:",
          "Key Responsibilities:",
          "- Collaborate closely with the lead developer to design system solutions.",
          "- Utilise existing group-wide infrastructure to optimise efficiency and avoid redundancy.",
          "- Take the lead in developing feature slices and efficiently allocate work within the team.",
          "- Provide mentorship to less experienced developers within the team.",
        ],
      },
      {
        position: "Developer",
        startDate: "Aug 2021",
        endDate: "Jun 2022",
        description: [],
      },
    ],
  },
  {
    company: {
      name: "AroFlo",
      linkedInUrl: "https://www.linkedin.com/company/aroflo",
    },
    positions: [
      {
        position: "Web Developer",
        startDate: "Aug 2019",
        endDate: "Sept 2021",
        description: [
          "Working as a full stack developer. Delivering new features to AroFlo Platform which is an industry leading product in Australia, my current main involves in the platform include but not limited to project management, quotes, invoices, payment integration areas.",
        ],
      },
    ],
  },
  {
    company: {
      name: "PTEPLUS",
    },
    positions: [
      {
        position: "Development Lead",
        startDate: "Jan 2018",
        endDate: "May 2019",
        description: [
          "Work with the dev team on two main products. Design and implement IT architecture on the clouding computing platforms (AWS and Aliyun). Deliver stable, rapid responsible and reliable web applications to product's users.",
        ],
      },
      {
        position: "Full Stack Developer",
        startDate: "Oct 2016",
        endDate: "Jan 2018",
        description: [
          "One of the key contributors to the start-up company, and participated in almost every stage of product design, implement and testing. Took responsibilities on full solutions to the technical side such as coding and servers' setup.",
        ],
      },
    ],
  },
];

export const backendStack: string[] = [
  "Node, TypeScript, JavaScript",
  "Java, ColdFusion",
  "PHP, Laravel, Moodle",
  "MySQL, MSSQL, Redis",
  "Server management, Message Queue",
  "Payment Integration",
];

export const frontendStack: string[] = [
  "HTML",
  "JavaScript, Vue, React, jQuery",
  "CSS, SASS, TailwindCSS, Styled Components, Bootstrap",
  "Webpack",
];

export const devOpsStack: string[] = [
  "AWS",
  "Infrastructure planning and implementing",
  "Horizontal scaling",
  "Serverless, Docker, ECS",
  "CI/CD, GitlabCI, CircleCI",
];
