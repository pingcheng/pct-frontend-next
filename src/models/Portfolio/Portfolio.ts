export type Portfolio = {
  slug: string;
  name: string;
  coverImage: string;
  url: string | null;
  shortDescription: string;
  longDescription: string;
  workplace: string;
  projectRole: string;
  roleDescription: string[];
  members: string[];
  screenshots: string[];
  hasScreenshots: boolean;
  year: string | null;
};

export function createPortfolio(slug: string, props: PortfolioProps): Portfolio {
  const screenshots = props.screenshots ?? [];
  return {
    slug,
    name: props.name ?? "",
    coverImage: props.coverImage ?? "",
    url: props.url ?? null,
    shortDescription: props.shortDescription ?? "",
    longDescription: props.longDescription ?? "",
    workplace: props.workplace ?? "",
    projectRole: props.projectRole ?? "",
    roleDescription: props.roleDescription ?? [],
    members: props.members ?? [],
    screenshots,
    hasScreenshots: screenshots.length > 0,
    year: props.year ?? null,
  };
}

export type PortfolioProps = {
  name: string | null;
  coverImage: string | null;
  url: string | null;

  shortDescription: string | null;
  longDescription: string | null;

  workplace: string | null;
  projectRole: string | null;
  roleDescription: string[] | null;

  members: string[] | null;
  screenshots: string[] | null;
  year: string | null;
};
