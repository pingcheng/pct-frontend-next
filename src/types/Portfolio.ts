export type { Portfolio } from "@/models/Portfolio/Portfolio";
export type { Portfolio as SerializedPortfolio } from "@/models/Portfolio/Portfolio";