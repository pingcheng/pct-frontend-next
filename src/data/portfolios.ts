import { createPortfolio } from "@/models/Portfolio/Portfolio";

export const Portfolios = [
  createPortfolio("empire-cbs", {
    name: "Empire CBS",
    shortDescription: "Empire CBS internal job management system",
    coverImage:
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/empire-cbs/empire-cbs.png",
    url: null,
    longDescription:
      "An internal job management system that helps Empire CBS to record and track their project progress and manage their resources, timesheets and etc.",
    workplace: "Private project",
    projectRole: "Backend & DevOps Developer",
    roleDescription: [
      "Backend developments",
      "CI/CD implementation",
      "DevOps and Server management",
    ],
    members: [
      "Ping Cheng - Backend & DevOps Developer",
      "Johnson Yao - Frontend developer",
    ],
    screenshots: [
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/empire-cbs/empire-cbs-1.png",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/empire-cbs/empire-cbs-2.png",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/empire-cbs/empire-cbs-3.png",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/empire-cbs/empire-cbs-4.png",
    ],
    year: "2020-2021",
  }),
  createPortfolio("pteplus", {
    name: "PTEPLUS",
    shortDescription:
      "PTEPLUS provides a user friendly UI and accurate machine scoring for PTE test taker",
    coverImage:
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/pteplus/pteplus.jpg",
    url: null,
    longDescription:
      "PTEPLUS is the first of its' kind 'smart platform' for English language practice in Australia. Its' Natural Language Processing technology provides PTE applicants with automated machine grading in alignment with the PTE Academic official score guide, such that PTE Academic future Test takers may practise for the Test, and also easily gauge their current standards and Test-readiness.",
    workplace: "Eduplus 2016 - 2019",
    projectRole: "Development Lead",
    roleDescription: [
      "The decision on tech stack",
      "Architecture design and implement",
      "Requirements analysis and breakdown",
      "Tasks split up and allocation",
      "Backend/Frontend developments",
      "Bridging different teams",
    ],
    members: [
      "Ping Cheng - Development lead",
      "Zoe Wang - Frontend developer",
      "Wayne Zhang - Backend developer",
      "Jeffery Wu - Backend developer",
      "Irene Zhang - Frontend developer",
    ],
    screenshots: [
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/pteplus/1.jpg",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/pteplus/2.jpg",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/pteplus/aio1.jpg",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/pteplus/aio2.jpg",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/pteplus/aio3.jpg",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/pteplus/aio4.jpg",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/pteplus/exam1.jpg",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/pteplus/op1.jpg",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/pteplus/op2.jpg",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/pteplus/op3.jpg",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/pteplus/op4.jpg",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/pteplus/op5.jpg",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/pteplus/report-analysis.jpg",
    ],
    year: "2016-2019",
  }),
  createPortfolio("pinhr", {
    name: "PinHR",
    shortDescription: "A customised human resources management tool",
    coverImage:
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/pinhr/cover.jpg",
    url: null,
    longDescription:
      "A customised human resources management tool, including online application form and auto email system",
    workplace: "EDUPLUS",
    projectRole: "Sole Developer",
    roleDescription: [
      "The decision on tech stack",
      "Architecture design and implement",
      "Requirements analysis and breakdown",
      "Backend/Frontend developments",
    ],
    members: ["Ping Cheng - Developer"],
    screenshots: [
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/pinhr/1.jpg",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/pinhr/2.jpg",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/pinhr/3.jpg",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/pinhr/4.jpg",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/pinhr/5.jpg",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/pinhr/6.jpg",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/pinhr/7.jpg",
    ],
    year: "2018",
  }),
  createPortfolio("vatgc", {
    name: "VAGTC",
    shortDescription: "The final year project in the university",
    coverImage:
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/vagtc/vagtc.png",
    url: "https://www.vagtc.org.au/",
    longDescription:
      "The final year project in the university, worked with the non-profit organisation to provide a modern and clean website. The web is WordPress based.",
    workplace: "Swinburne University",
    projectRole: "Developer",
    roleDescription: [
      "Setup the WordPress",
      "Customise the theme",
      "Integrate plugins",
    ],
    members: [
      "Fame - Team Leader",
      "Ping Cheng - Developer",
      "Salil Batra - Developer",
    ],
    screenshots: [
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/vagtc/1.jpg",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/vagtc/2.jpg",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/vagtc/3.jpg",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/vagtc/4.jpg",
    ],
    year: "2015",
  }),
  createPortfolio("qce-eduplus", {
    name: "QCE Eduplus",
    shortDescription:
      "Maths practice platform partners with Cengage for Queensland secondary schools",
    coverImage:
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/qce-eduplus/cover.jpg",
    url: null,
    longDescription:
      "EduPlus empowers teachers and students with immediate feedback, live progress tracking, and real-time actionable insights â€“ so you can harness more potential with less effort",
    workplace: "EDUPLUS",
    projectRole: "Development Lead",
    roleDescription: [
      "The decision on tech stack",
      "Architecture design and implement",
      "Requirements analysis and breakdown",
      "Tasks split up and allocation",
      "Backend/Frontend developments",
      "Bridging different teams",
    ],
    members: [
      "Ping Cheng - Development Lead",
      "Jeffery Wu - Backend/Frontend developer",
    ],
    screenshots: [
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/qce-eduplus/1.jpg",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/qce-eduplus/2.jpg",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/qce-eduplus/3.jpg",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/qce-eduplus/4.jpg",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/qce-eduplus/5.jpg",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/qce-eduplus/6.jpg",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/qce-eduplus/7.jpg",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/qce-eduplus/8.jpg",
    ],
    year: "2019-2020",
  }),
  createPortfolio("rhfoto", {
    name: "RH.foto",
    shortDescription:
      "A simplistic and clean photo wall for a group of professional photographers",
    coverImage:
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/rhfoto/rhfoto.png",
    url: "https://rhfoto.com.au/",
    longDescription:
      "A simple and clean Wordpress based website for a group of professional wedding photographers demonstrates their abilities to deliver high end, breathtaking, fashionable and personalised images.",
    workplace: "Private project",
    projectRole: "Sole Developer",
    roleDescription: [
      "Wordpress customisation",
      "Theme development",
      "Plugin integration",
    ],
    members: ["Ping Cheng - Developer"],
    screenshots: [
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/rhfoto/1.jpg",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/rhfoto/2.jpg",
      "https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/rhfoto/3.jpg",
    ],
    year: "2017",
  }),
];
