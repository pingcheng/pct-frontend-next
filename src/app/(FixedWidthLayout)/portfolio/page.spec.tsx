import { render, screen } from "@testing-library/react";
import Portfolio, { metadata } from "./page";
import { Portfolios } from "@/data/portfolios";

describe("Portfolio page", () => {
  it("matches snapshot", () => {
    const { container } = render(<Portfolio />);
    expect(container).toMatchSnapshot();
  });

  it("should display the correct heading", () => {
    render(<Portfolio />);
    const heading = screen.getByRole("heading", { level: 1 });
    expect(heading).toHaveTextContent("Portfolio");
  });

  it("should display all portfolio items with correct links", () => {
    render(<Portfolio />);

    Portfolios.forEach((portfolio) => {
      // Check if portfolio card link exists with correct href
      const links = screen.getAllByRole("link");
      const portfolioLink = links.find(
        (link) => link.getAttribute("href") === `/portfolio/${portfolio.slug}`
      );
      expect(portfolioLink).toBeInTheDocument();

      // Check if portfolio card exists
      const card = portfolioLink?.querySelector(".portfolioCard");
      expect(card).toBeInTheDocument();

      // Check if portfolio details are rendered
      const name = portfolioLink?.querySelector(".text-sm.font-bold");
      const descriptions = portfolioLink?.querySelectorAll(".text-xs");
      const shortDescription = descriptions?.[1]; // Second .text-xs element is the description
      expect(name).toHaveTextContent(portfolio.name);
      expect(shortDescription).toHaveTextContent(portfolio.shortDescription);
    });
  });

  it("should have correct metadata", () => {
    expect(metadata.title).toBe("Portfolio | Ping Cheng - Software Development Projects");
  });
});
