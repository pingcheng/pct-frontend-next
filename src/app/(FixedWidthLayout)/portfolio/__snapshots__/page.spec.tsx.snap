// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Portfolio page matches snapshot 1`] = `
<div>
  <div
    class="py-4"
  >
    <div
      class="animate-fade-in"
    >
      <h1
        class="heading1 center"
      >
        Portfolio
      </h1>
    </div>
    <div
      class="mt-4 flex flex-row flex-wrap gap-8 justify-center"
    >
      <div
        class="animate-fade-in-scale animate-delay-200"
      >
        <a
          href="/portfolio/empire-cbs"
        >
          <div
            class="perspectiveContainer"
          >
            <div
              aria-label="Empire CBS project"
              class="portfolioCard"
              role="article"
              style="background-image: url("https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/empire-cbs/empire-cbs.png"); background-size: cover; transform: rotateX(0deg) rotateY(0deg);"
            >
              <div
                class="absolute bottom-0 p-2 text-gray-200"
                style="z-index: 10;"
              >
                <div
                  class="flex items-center gap-2"
                >
                  <div
                    class="text-sm font-bold"
                  >
                    Empire CBS
                  </div>
                  <div
                    class="text-xs text-gray-300"
                  >
                    (
                    2020-2021
                    )
                  </div>
                </div>
                <div
                  class="text-xs"
                >
                  Empire CBS internal job management system
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
      <div
        class="animate-fade-in-scale animate-delay-300"
      >
        <a
          href="/portfolio/pteplus"
        >
          <div
            class="perspectiveContainer"
          >
            <div
              aria-label="PTEPLUS project"
              class="portfolioCard"
              role="article"
              style="background-image: url("https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/pteplus/pteplus.jpg"); background-size: cover; transform: rotateX(0deg) rotateY(0deg);"
            >
              <div
                class="absolute bottom-0 p-2 text-gray-200"
                style="z-index: 10;"
              >
                <div
                  class="flex items-center gap-2"
                >
                  <div
                    class="text-sm font-bold"
                  >
                    PTEPLUS
                  </div>
                  <div
                    class="text-xs text-gray-300"
                  >
                    (
                    2016-2019
                    )
                  </div>
                </div>
                <div
                  class="text-xs"
                >
                  PTEPLUS provides a user friendly UI and accurate machine scoring for PTE test taker
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
      <div
        class="animate-fade-in-scale animate-delay-400"
      >
        <a
          href="/portfolio/pinhr"
        >
          <div
            class="perspectiveContainer"
          >
            <div
              aria-label="PinHR project"
              class="portfolioCard"
              role="article"
              style="background-image: url("https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/pinhr/cover.jpg"); background-size: cover; transform: rotateX(0deg) rotateY(0deg);"
            >
              <div
                class="absolute bottom-0 p-2 text-gray-200"
                style="z-index: 10;"
              >
                <div
                  class="flex items-center gap-2"
                >
                  <div
                    class="text-sm font-bold"
                  >
                    PinHR
                  </div>
                  <div
                    class="text-xs text-gray-300"
                  >
                    (
                    2018
                    )
                  </div>
                </div>
                <div
                  class="text-xs"
                >
                  A customised human resources management tool
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
      <div
        class="animate-fade-in-scale animate-delay-500"
      >
        <a
          href="/portfolio/vatgc"
        >
          <div
            class="perspectiveContainer"
          >
            <div
              aria-label="VAGTC project"
              class="portfolioCard"
              role="article"
              style="background-image: url("https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/vagtc/vagtc.png"); background-size: cover; transform: rotateX(0deg) rotateY(0deg);"
            >
              <div
                class="absolute bottom-0 p-2 text-gray-200"
                style="z-index: 10;"
              >
                <div
                  class="flex items-center gap-2"
                >
                  <div
                    class="text-sm font-bold"
                  >
                    VAGTC
                  </div>
                  <div
                    class="text-xs text-gray-300"
                  >
                    (
                    2015
                    )
                  </div>
                </div>
                <div
                  class="text-xs"
                >
                  The final year project in the university
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
      <div
        class="animate-fade-in-scale animate-delay-600"
      >
        <a
          href="/portfolio/qce-eduplus"
        >
          <div
            class="perspectiveContainer"
          >
            <div
              aria-label="QCE Eduplus project"
              class="portfolioCard"
              role="article"
              style="background-image: url("https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/qce-eduplus/cover.jpg"); background-size: cover; transform: rotateX(0deg) rotateY(0deg);"
            >
              <div
                class="absolute bottom-0 p-2 text-gray-200"
                style="z-index: 10;"
              >
                <div
                  class="flex items-center gap-2"
                >
                  <div
                    class="text-sm font-bold"
                  >
                    QCE Eduplus
                  </div>
                  <div
                    class="text-xs text-gray-300"
                  >
                    (
                    2019-2020
                    )
                  </div>
                </div>
                <div
                  class="text-xs"
                >
                  Maths practice platform partners with Cengage for Queensland secondary schools
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
      <div
        class="animate-fade-in-scale animate-delay-700"
      >
        <a
          href="/portfolio/rhfoto"
        >
          <div
            class="perspectiveContainer"
          >
            <div
              aria-label="RH.foto project"
              class="portfolioCard"
              role="article"
              style="background-image: url("https://pct-frontend-assets.s3.ap-southeast-2.amazonaws.com/portfolio/rhfoto/rhfoto.png"); background-size: cover; transform: rotateX(0deg) rotateY(0deg);"
            >
              <div
                class="absolute bottom-0 p-2 text-gray-200"
                style="z-index: 10;"
              >
                <div
                  class="flex items-center gap-2"
                >
                  <div
                    class="text-sm font-bold"
                  >
                    RH.foto
                  </div>
                  <div
                    class="text-xs text-gray-300"
                  >
                    (
                    2017
                    )
                  </div>
                </div>
                <div
                  class="text-xs"
                >
                  A simplistic and clean photo wall for a group of professional photographers
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
